apps/backend/src/modules/users/user.service.ts

```diff
@@ -43,33 +48,45 @@ class UserService implements Service {
 			}),
 		);
 
-		return item.toObject();
+		if (!item.toObject().id) {
+			throw new UserError({
+				message: CommonExceptionMessage.COMMON_EXCEPTION_MESSAGE,
+				status: HTTPCode.INTERNAL_SERVER_ERROR,
+			});
+		}
+
+		const newUser = await this.findById(item.toObject().id);
+
+		if (!newUser) {
+			throw new UserError({
+				message: CommonExceptionMessage.COMMON_EXCEPTION_MESSAGE,
+				status: HTTPCode.NOT_FOUND,
+			});
+		}
+
+		return newUser.toObject() as UserAuthResponseDto;
```

Comment:
not sure what is this for, create should return the needed interface
```suggestion
		return item.toObject()
```